architecture/builds.adoc

```diff
@@ -44,3 +44,53 @@ STI builds are a replacement for the OpenShift v2-like developer experience. The
 == Custom Build
 Custom builds are the most sophisticated version of builds, allowing developers to define a builder image which is responsible for the entire process of the build. The custom builder image is a plain Docker image within which the author embeds the logic of the desired build process, such as building RPMs or building
 base Docker images.
+
+[#using-docker-credentials-for-pushing-images]
+== Using Docker credentials for pushing images
+
+In case you want to push the output image into private Docker Registry that
+requires authentication or Docker Hub, you have to supply the `dockercfg` file
+with valid Docker Registry credentials.
+
+The `.dockercfg` JSON file usually exists in your home directory and it has following
+format:
+
+```
+{"https://index.docker.io/v1/":{"auth":"encrypted_password","email":"foo@bar.com"}}
+```
+
+You can also create this file by executing `docker login` command.
```

Comment:
@soltysh if it exists then you can use this command to add multiple entries to the file. I think the current text is ok.